extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -1126,10 +1126,9 @@ export class AmpStoryPage extends AMP.BaseElement {
     const navigationPath = this.storeService_.get(
       StateProperty.NAVIGATION_PATH
     );
-    // Navigation path is a complete list of the navigation history, including
-    // the currently active page. The previous page is the next to last element
-    // of that array.
-    const previousPageId = navigationPath[navigationPath.length - 2];
+
+    const currentPageIndex = navigationPath.indexOf(this.element.id);
+    const previousPageId = navigationPath[currentPageIndex - 1];
```

Comment:
This is not correct: with branching, you could go twice through the same page. In which case, when navigating back, you'd always get the previous index of the first navigation.

My suggestion:
Either
- if the page is `active`, keep the current code
- if not `active`, add your code with a reverse loop search on the `navigationPath` array instead of the `indexOf`

or

- reverse loop search on the `navigationPath` array in all cases, since it'd be only one iteration of the loop for the general case